pkg/cache/v3/simple.go

```diff
@@ -737,20 +754,27 @@ func createDeltaResponse(snapshot ResourceSnapshot, watch DeltaResponseWatch, re
 	}
 
 	for name := range returnedResources {
-		_, exist := resources[name]
+		res, exist := resources[name]
 		if !exist {
 			deletedResources = append(deletedResources, name)
 			delete(returnedResources, name)
 			continue
 		}
 
 		if watch.subscription.IsWildcard() {
-			continue
-		}
-		if _, watched := subscribed[name]; !watched {
-			// Resource is no longer watched.
-			deletedResources = append(deletedResources, name)
-			delete(returnedResources, name)
+			// For wildcard subscriptions: delete if resource is on-demand only AND not explicitly subscribed
+			if res.IsOnDemandOnly() {
```

Comment:
Minor: could this be simplified in a simple check?
Stg like if isWildcard and !ondemand, continue, else check for subscribe below